src/sagas.js

```diff
@@ -525,10 +524,7 @@ function* fetchAllTemplates (action) {
 
   if (templates && templates['template']) {
     const templatesInternal = templates.template.map(template => Api.templateToInternal({ template }))
-    yield put(addTemplates({ templates: templatesInternal }))
-
-    const templateIdsToPreserve = templatesInternal.map(item => item.id)
-    yield put(removeMissingTemplates({ templateIdsToPreserve }))
+    yield put(setTemplates(templatesInternal))
```

Comment:
I like SET instead of ADD/REMOVE for templates (and for the other reducers like templates that are really only set by refreshData.